infrastructure/smart-contracts/contracts/upgrades/referee/Referee9.sol

```diff
@@ -1088,40 +907,121 @@ contract Referee9 is Initializable, AccessControlEnumerableUpgradeable {
     }
 
     /** 
-    * @notice Submit Pool Assertion external function
-    * @dev this function is called by the pool owner, or an approved operator, to submit a pool assertion to a challenge.
-    * @param _poolAddress The address of the pool.
+    * @notice Submit Bulk Assertion
+    * @dev this function is called by a pool owner, an approved operator, or a wallet woner with unstaked keys to submit a bulk assertion to a challenge.
+    * @param _bulkAddress The address of the pool or owner wallet.
     * @param _challengeId The ID of the challenge.
     * @param _confirmData The confirm data of the assertion.
     */
-    function submitPoolAssertion(address _poolAddress, uint256 _challengeId, bytes memory _confirmData) external {
+    function submitBulkAssertion(address _bulkAddress, uint256 _challengeId, bytes memory _confirmData) external {
         // Confirm the challenge is open for submissions
 		require(challenges[_challengeId].openForSubmissions, "16");
 
+        // Initially set the number of keys to the number of keys assigned to the pool (will be 0 if pool does not exist)
+        uint256 numberOfKeys = assignedKeysToPoolCount[_bulkAddress] + assignedKeysOfUserCount[_bulkAddress];
+
         // Check if the submission successor hash, doesn't match the one submitted by the challenger, then end early and emit an event
 		if (keccak256(abi.encodePacked(_confirmData)) != keccak256(abi.encodePacked(challenges[_challengeId].assertionStateRootOrConfirmData))) {
-            emit InvalidBatchSubmission(_challengeId, msg.sender, assignedKeysToPoolCount[_poolAddress]);
+            emit InvalidBatchSubmission(_challengeId, msg.sender, numberOfKeys);
```

Comment:
We should update this event if its not used in the subgraph.
It should also emit the owner / pool address the msg.sender tried to submit for.

Reply:
Done!